### PR TITLE
fix: allow kubelet feature-gate overrides

### LINKED ISSUE #4112: aks-engine applies default feature gates instead of honoring user-defined feature gates
**Describe the bug**

I explictly set `ExecProbeTimeout=true` in kubelet feature gates and run `aks-engine generate` but the generated API model does not reflect that:

```json 

        "kubeletConfig": {
          ...
          "--feature-gates": "ExecProbeTimeout=true,KubeletPodResources=false"
          ...
```

The generated API model has the following feature gates:

```json
        "kubeletConfig": {
          ...
          "--feature-gates": "ExecProbeTimeout=false,KubeletPodResources=false,RotateKubeletServerCertificate=true",
          ...
```

**Expected behavior**

```json
        "kubeletConfig": {
          ...
          "--feature-gates": "ExecProbeTimeout=true,KubeletPodResources=false,RotateKubeletServerCertificate=true",
          ...
```

**AKS Engine version**

v0.58.0

**Kubernetes version**

v1.20

**Additional context**

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This PR ensures that user-override values of the kubelet --feature-gates interface are respected.

tl;dr fixes a nasty bug in the way that feature-gates key/val pairs were composed, we are currently throwing away the user-provided input if there's a default value for that key

**Issue Fixed**:

Fixes #4112

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jsturtevant** [comment]
looks like a duplicate of #4113
